helix-term/src/keymap.rs

```diff
@@ -198,13 +198,44 @@ impl<'de> serde::de::Visitor<'de> for KeyTrieVisitor {
     where
         M: serde::de::MapAccess<'de>,
     {
+        let mut label = "";
+        let mut command = None;
         let mut mapping = HashMap::new();
         let mut order = Vec::new();
-        while let Some((key, value)) = map.next_entry::<KeyEvent, KeyTrie>()? {
-            mapping.insert(key, value);
-            order.push(key);
+
+        while let Some(key) = map.next_key::<&str>()? {
+            match key {
+                "label" => label = map.next_value::<&str>()?,
+                "command" => command = Some(map.next_value::<MappableCommand>()?),
+                _ => {
+                    let key_event = key.parse::<KeyEvent>().map_err(serde::de::Error::custom)?;
+                    let key_trie = map.next_value::<KeyTrie>()?;
+                    mapping.insert(key_event, key_trie);
+                    order.push(key_event);
+                }
+            }
+        }
+
+        match command {
+            None => Ok(KeyTrie::Node(KeyTrieNode::new(label, mapping, order))),
+            Some(cmd) => {
+                if label.is_empty() {
+                    Ok(KeyTrie::Leaf(cmd))
+                } else {
+                    match cmd {
+                        MappableCommand::Typable { name, args, .. } => {
+                            Ok(MappableCommand::Typable {
+                                name,
+                                args,
+                                doc: label.to_string(),
+                            })
+                            .map(KeyTrie::Leaf)
+                        }
+                        MappableCommand::Static { .. } => Ok(KeyTrie::Leaf(cmd)),
```

Comment:
I don't like the duplicate `Ok(KeyTrie::Leaf(cmd))` here. It feels like it should be possible to do this in a single `let if`, but I keep running into used after move issues that I haven't been able to sidestep.

Reply:
...and of course I figure it out 5 minutes after opening the PR. Thanks, PR duck! :duck: